File: Uler/Patient.py
from Person import Person
from Medicine import Medicine

class Patient(Person):
    def __init__(self, name: str, disease: str):
        super().__init__(name)
        self.disease = disease
        self.medicines: list[Medicine] = []

    #get set
    def set_disease(self, disease: str):
        self.disease = disease

    def get_disease(self) -> str:
        return self.disease

    #fungsi add obat
    def add_medicine(self, m: Medicine):
        self.medicines.append(m)

    #fungsi print pasien
    def print_patient(self):
        print(f"Patient: {self.get_name()}, Disease: {self.disease}")
        print(" Medicines:")
        if not self.medicines:
            print("- No medicines prescribed.")
        else:
            for m in self.medicines:
                m.print_medicine()
        print("================================================")

File: CPP/Code/Patient.cpp
#include <iostream>
#include <vector>
using namespace std;

class Patient : public Person {
private:
    string disease;
    vector<Medicine> medicines;

public:
    Patient(string name, string disease) : Person(name), disease(disease) {}

    //get set
    void setDisease(string disease) { 
        this->disease = disease; 
    }
    
    string getDisease() const { 
        return disease; 
    }

    //fungsi add obat
    void addMedicine(const Medicine& m) {
        medicines.push_back(m);
    }

    //fungsi print pasien
    void printPatient() {
        cout << "Patient: " << getName() << ", Disease: " << disease << endl;
        cout << " Medicines:" << endl;
        if (medicines.empty()) {
            cout << "- No medicines prescribed." << endl;
        } else {
            for (auto& m : medicines) {
                m.printMedicine();
            }
        }
        cout << "================================================" << endl; // separator
    }
};